.github/workflows/pr-one-shot-test.yml

```diff
@@ -0,0 +1,45 @@
+---
+name: PR One-Shot Test
+
+on:
+  pull_request_review:
+    types: ["submitted"]
+
+jobs:
+  one-shot-test:
+    # Only run when a pull request is approved
+    if: github.event.review.state == 'approved'
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          submodules: 'recursive'
+
+      - name: Set up Python 3.13
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.13'
+
+      - name: Install uv
+        run: |
+          curl -LsSf https://astral.sh/uv/install.sh | sh
+          echo "$HOME/.cargo/bin" >> $GITHUB_PATH
+
+      - name: Run one-shot test
+        env:
+          # OpenAI configuration
+          OPENAI_API_KEY: ${{ secrets.OPENAI_API_KEY || 'sk-test-placeholder-key-for-ci' }}
+          OPENAI_BASE_URL: ${{ secrets.OPENAI_BASE_URL || '' }}
+          OPENAI_MODEL_NAME: ${{ secrets.OPENAI_MODEL_NAME || 'gpt-4o' }}
+          OPENAI_TEMPERATURE: ${{ secrets.OPENAI_TEMPERATURE || '0.7' }}
+          # Server configuration
+          PORT: ${{ secrets.PORT || '8080' }}
+          HOST: ${{ secrets.HOST || '0.0.0.0' }}
+          LOG_LEVEL: ${{ secrets.LOG_LEVEL || 'INFO' }}
+          # Application configuration
+          LOCAL_TOOLS_ENABLED: ${{ secrets.LOCAL_TOOLS_ENABLED || 'true' }}
+          MAX_TURNS: ${{ secrets.MAX_TURNS || '25' }}
```

Comment:
these aren't secrets, just environment variables, only OPENAI_API_KEY is a secret